text/0000-named-blocks.md

```diff
@@ -0,0 +1,727 @@
+  </@header>
+
+  <@body as |foo|>
+    Body {{foo}}
+  </@body>
+</x-modal>
+```
+
+As demonstrated above, the _nested_ syntax for both curly and
+angle-bracket multi-block syntax has the format `<@blockName>...</@blockName>`.
+
+This multi-block syntax cannot be mixed with other syntaxes; either ALL
+the nested "children" of a component invocation need to be
+`<@blockName>...</@blockName>` (multi-block syntax), or none of them do
+(classic single-block syntax). The presence of any non-whitespace
+character between or around `<@blockName>...</@blockName>`s is a
+compile-time error.
+
+Passing two blocks with the same name is a compiler error (though this
+might be relaxed in a future RFC).
+
+### Blocks are just (opaque) data
+
+This RFC introduces the concept that blocks are just opaque values
+passed into components as data, rather than living in what is
+essentially a separate namespace only accessible to `{{yield}}`.
+
+In the above example (with curly syntax), Ember Component `x-foo` would
+have its `header` and `body` properties set on its instance. This means,
+among other things, that there's no need for a
+[hasBlock API for JavaScript](https://github.com/emberjs/rfcs/pull/102);
+you can just use normal property lookup / computed properties / etc to
+determine whether a block is provided. This means that blocks can
+be stashed on services and rendered elsewhere, e.g. the
+[ember-elsewhere use case](https://github.com/emberjs/rfcs/pull/226#issuecomment-299949000).
+
+### Unified Renderable Syntax
+
+Rather than continuing to enhance the `{{yield}}` syntax, we should take
+this opportunity to unify the various syntaxes for rendering things,
+from blocks to primitive values to component factories.
+
+We'll use the following example component invocation to explore
+what this syntax looks like: the following (curly syntax) invocation is valid syntax for
+rendering either an Ember.Component or a Glimmer Component named
+`x-modal` and passing it 3 named blocks: `header`, `body`, and `footer`:
+
+```html
+{{#x-modal}}
+  <@header as |title|>
+    Header {{title}}
+  </@header>
+
+  <@body>
+    Body
+  </@body>
+
+  <@footer>
+    Footer
+  </@footer>
+{{/x-modal}}
+```
+
+Given the above invocation, here's how you could render these blocks:
+
+```
+{{! within ember-component layout }}
+{{this.header "ima title"}}
+{{this.body}}
+{{this.footer}}
+
+{{! within glimmer-component layout }}
+{{@header "ima title"}}
+{{@body}}
+{{@footer}}
+```
+
+Both of these Ember/Glimmer layouts would render:
+
+```
+Header ima title
+Body
+Footer
+```
+
+The mental modal here is that is that for ECs, named blocks are
+set/bound as a properties on the instance, which we're rendering the
+same way we always rendering properties on the instance. For GCs, blocks
+are just args that we're rendering with the standard @arg syntax
+
+#### Why `{{this.header}}` and not just `{{header}}`?
+
+Unfortunately, we are constrained by the fact that `{{foo}}` means:
+
```

Comment:
Not sure if it applies here, but there is one more rule to the list below;

- Look for a block param named `foo`, and render it

That one "wins" over all, per: https://github.com/emberjs/ember.js/issues/14413#issuecomment-251666245